rutil/Errdes.hxx

```diff
@@ -0,0 +1,268 @@
+/*EPROTONOSUPPORT 93  */  "EPROTONOSUPPORT (Protocol not supported)",
+/*ESOCKTNOSUPPORT 94  */  "ESOCKTNOSUPPORT (Socket type not supported)",
+/*EOPNOTSUPP      95  */  "EOPNOTSUPP (Operation not supported)",
+/*EPFNOSUPPORT    96  */  "EPFNOSUPPORT (Protocol family not supported)",
+/*EAFNOSUPPORT    97  */  "EAFNOSUPPORT (Address family not supported by protocol)",
+/*EADDRINUSE      98  */  "EADDRINUSE (Address already in use)",
+/*EADDRNOTAVAIL   99  */  "EADDRNOTAVAIL (Cannot assign requested address)",
+/*ENETDOWN        100 */  "ENETDOWN (Network is down)",
+/*ENETUNREACH     101 */  "ENETUNREACH (Network is unreachable)",
+/*ENETRESET       102 */  "ENETRESET (Network dropped connection on reset)",
+/*ECONNABORTED    103 */  "ECONNABORTED (Software caused connection abort)",
+/*ECONNRESET      104 */  "ECONNRESET (Connection reset by peer)",
+/*ENOBUFS         105 */  "ENOBUFS (No buffer space available)",
+/*EISCONN         106 */  "EISCONN (Transport endpoint is already connected)",
+/*ENOTCONN        107 */  "ENOTCONN (Transport endpoint is not connected)",
+/*ESHUTDOWN       108 */  "ESHUTDOWN (Cannot send after transport endpoint shutdown)",
+/*ETOOMANYREFS    109 */  "ETOOMANYREFS (Too many references: cannot splice)",
+/*ETIMEDOUT       110 */  "ETIMEDOUT (Connection timed out)",
+/*ECONNREFUSED    111 */  "ECONNREFUSED (Connection refused)",
+/*EHOSTDOWN       112 */  "EHOSTDOWN (Host is down)",
+/*EHOSTUNREACH    113 */  "EHOSTUNREACH (No route to host)",
+/*EALREADY        114 */  "EALREADY (Operation already in progress)",
+/*EINPROGRESS     115 */  "EINPROGRESS (Operation now in progress)",
+/*ESTALE          116 */  "ESTALE (Stale file handle)",
+/*EUCLEAN         117 */  "EUCLEAN (Structure needs cleaning)",
+/*ENOTNAM         118 */  "ENOTNAM (Not a XENIX named type file)",
+/*ENAVAIL         119 */  "ENAVAIL (No XENIX semaphores available)",
+/*EISNAM          120 */  "EISNAM (Is a named type file)",
+/*EREMOTEIO       121 */  "EREMOTEIO (Remote I/O error)",
+/*EDQUOT          122 */  "EDQUOT (Disk quota exceeded)",
+/*ENOMEDIUM       123 */  "ENOMEDIUM (No medium found)",
+/*EMEDIUMTYPE     124 */  "EMEDIUMTYPE (Wrong medium type)",
+/*ECANCELED       125 */  "ECANCELED (Operation canceled)",
+/*ENOKEY          126 */  "ENOKEY (Required key not available)",
+/*EKEYEXPIRED     127 */  "EKEYEXPIRED (Key has expired)",
+/*EKEYREVOKED     128 */  "EKEYREVOKED (Key has been revoked)",
+/*EKEYREJECTED    129 */  "EKEYREJECTED (Key was rejected by service)",
+/*EOWNERDEAD      130 */  "EOWNERDEAD (Owner died)",
+/*ENOTRECOVERABLE 131 */  "ENOTRECOVERABLE (State not recoverable)",
+/*ERFKILL         132 */  "ERFKILL (Operation not possible due to RF-kill)",
+/*EHWPOISON       133 */  "EHWPOISON (Memory page has hardware error)",
+};
+static const int NUM_MESSAGES_OS = sizeof(messagesOS)/sizeof(messagesOS[0]);
+
+static inline char* errortostringOS(int errnumOS)
+{
+   if (errnumOS < NUM_MESSAGES_OS)
```

Comment:
what if errnumOS is a negative value?  Should the function expected an unsigned value or should the range check ensure it is >= 0?